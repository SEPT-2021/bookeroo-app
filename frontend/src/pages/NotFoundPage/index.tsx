import React from "react";
import {
  Button,
  createStyles,
  Grid,
  Link,
  Theme,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import logo from "../../assets/logo.svg";

function NotFoundPage({ classes }: PageNotFound) {
  return (
    <Grid container component="main" className={classes.root}>
      <Grid
        container
        spacing={0}
        direction="column"
        alignItems="center"
        justify="center"
        style={{ minHeight: "100vh" }}
      >
        <img src={logo} alt="logo" />

        <h1>404 - Not Found!</h1>
        <Link href="/">
          <Button variant="contained" color="primary">
            Go Home
          </Button>
        </Link>
      </Grid>
    </Grid>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    root: {
      height: "100vh",
    },
    steps: {
      "& img": {
        width: 500,
        margin: theme.spacing(6),
      },
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      flexDirection: "column",
    },
    paper: {
      margin: theme.spacing(8, 4),
      display: "flex",
      flexDirection: "column",
      alignItems: "center",
    },
    avatar: {
      margin: theme.spacing(1),
      backgroundColor: theme.palette.secondary.main,
    },
    submit: {
      margin: theme.spacing(3, 0, 2),
    },
  });

type PageNotFound = WithStyles<typeof styles>;

export default withStyles(styles)(NotFoundPage);
