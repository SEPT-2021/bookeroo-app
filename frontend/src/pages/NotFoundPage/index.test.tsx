import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import { Button } from "@material-ui/core";
import { QueryClient, QueryClientProvider } from "react-query";
import NotFoundPage from "./index";

afterEach(cleanup);
describe("Rendering page NotFoundPage", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<NotFoundPage />, div);
    expect(div).toMatchSnapshot();
  });
});

describe("Button render matches snapshot", () => {
  test("renders button", () => {
    const button = render(<Button />);
    expect(button).toMatchSnapshot();
  });
});

describe("NotFoundPage matches snapshot", () => {
  test("render snapshot", () => {
    const { container } = render(
      <QueryClientProvider client={new QueryClient()}>
        <NotFoundPage />
      </QueryClientProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
