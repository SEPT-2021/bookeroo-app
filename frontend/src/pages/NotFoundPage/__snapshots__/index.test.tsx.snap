// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render matches snapshot renders button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`NotFoundPage matches snapshot matches snapshot 1`] = `
<div>
  <main
    class="MuiGrid-root NotFoundPage-root-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      style="min-height: 100vh;"
    >
      <img
        alt="logo"
        src="logo.svg"
      />
      <h1>
        404 - Not Found!
      </h1>
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="/"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Go Home
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </main>
</div>
`;

exports[`NotFoundPage matches snapshot render snapshot 1`] = `
<div>
  <main
    class="MuiGrid-root NotFoundPage-root-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      style="min-height: 100vh;"
    >
      <img
        alt="logo"
        src="logo.svg"
      />
      <h1>
        404 - Not Found!
      </h1>
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="/"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Go Home
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </main>
</div>
`;

exports[`Rendering page NotFoundPage renders without crashing 1`] = `
<div>
  <main
    class="MuiGrid-root NotFoundPage-root-1 MuiGrid-container"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-center"
      style="min-height: 100vh;"
    >
      <img
        alt="logo"
        src="logo.svg"
      />
      <h1>
        404 - Not Found!
      </h1>
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="/"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Go Home
          </span>
        </button>
      </a>
    </div>
  </main>
</div>
`;
